### PR TITLE
Update the CLI to work with the updated APIs format

### PR DESCRIPTION
**Pull request by federicoweber**
APIs changes in https://github.com/trunk-io/trunk/pull/20034

### CODE REVIEW

File: bundle/src/types.rs [hunk 1]
```diff
@@ -20,7 +21,7 @@ pub struct QuarantineRunResult {
     pub quarantine_status: QuarantineBulkTestStatus,
 }
 
-#[derive(Debug, Serialize, Clone, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Serialize, Clone, Deserialize, PartialEq, Eq, Hash)] // Add `Hash` here
```
Comment (dfrankland) [inline bundle/src/types.rs:24]: Is this needed? Seems like we're just sticking `Test` in `Vec`

File: bundle/src/lib.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@ mod bundle_meta;
 mod bundler;
 mod custom_tag;
 mod files;
-mod types;
+pub mod types;
```
Comment (dfrankland) [inline bundle/src/lib.rs:5]: Not needed, we already `pub use types::*;` down below

File: api/src/lib.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
-use std::collections::HashSet;
-
+use bundle::types::Test;
 use context::repo::RepoUrlParts;
 use serde::{Deserialize, Serialize};
+use std::collections::HashSet; // Add this import
```
Comment (dfrankland) [inline api/src/lib.rs:4]: Can we undo the change to `std::collections::HashSet` here? It was more nicely organized before

File: cli-tests/src/upload.rs [hunk 4]
```diff
@@ -51,17 +48,27 @@ async fn upload_bundle() {
...
-            repo: Repo {
-                host: String::from("github.com"),
-                owner: String::from("trunk-io"),
-                name: String::from("analytics-cli"),
-            },
-            org_url_slug: String::from("test-org"),
-        })
-    );
+    let quarantine_request = requests_iter.next().unwrap();
+    if let RequestPayload::GetQuarantineBulkTestStatus(req) = quarantine_request {
```
Comment (dfrankland) [inline cli-tests/src/upload.rs:52]: nit: use `assert_matches!()` for this. There's an example below on how to use it

File: cli/src/api_client/mod.rs [hunk 5]
```diff
@@ -159,10 +160,14 @@ impl ApiClient {
...
-                    .await
-                    .context("Failed to get response body as json.")
+                let response_body = response.text().await?;
+                match serde_json::from_str::<api::QuarantineConfig>(&response_body) {
+                    Ok(config) => Ok(config),
+                    Err(e) => {
+                        log::error!("Error retrieving quarantine configuration");
+                        return Err(anyhow::anyhow!("Failed to parse response body as json: {}", e));
+                    }
+                }
```
Comment (dfrankland) [inline cli/src/api_client/mod.rs:170]: No need for this change. This is effectively the same thing, but much more verbose

File: cli/src/api_client/mod.rs [hunk 6]
```diff
@@ -172,7 +177,14 @@ impl ApiClient {
...
         .call_api()
-        .await
+        .await;
+
+        if let Err(ref e) = result {
+            log::error!("Error during API call: {}", e);
+            return Err(anyhow::anyhow!(e.to_string()));
+        }
+
+        result
```
Comment (dfrankland) [inline cli/src/api_client/mod.rs:187]: nit: remove some redundancy

```suggestion
        if result.is_err() {
            log::error!("Error retrieving quarantine configuration");
        }

        result
```

File: cli/src/api_client/mod.rs (hunk 5 above)
Comment (federicoweber) [reply cli/src/api_client/mod.rs:170]: ah sorry forgot to clean this up, I broke it down for debugging to log the body

**Review by dfrankland — APPROVED** [review]
